bin/dart

```diff
@@ -43,8 +43,14 @@ function follow_links() (
   echo "$file"
 )
 
-PROG_NAME="$(follow_links "$BASH_SOURCE")"
+PROG_NAME="$(follow_links "${BASH_SOURCE[0]}")"
 BIN_DIR="$(cd "${PROG_NAME%/*}" ; pwd -P)"
+OS="$(uname -s)"
+
+if [[ $OS =~ MINGW.* ]]; then
+  # If we're on Windows, invoke the batch script instead, to get proper locking.
+  exec "${BIN_DIR}/dart.bat" "$@"
```

Comment:
do we want to exit after this?

Reply:
The [`exec` command](https://www.computerhope.com/unix/bash/exec.htm) completely substitutes the given command for the current process, so there's no need to exit (it already has exited by invoking the other script).

Reply:
Also, the final return value is the return value of the other script.

Reply:
cool

Reply:
I guess that means we should never reach the other switch cases on MINGW anyway?

Reply:
What other switch cases?  I mean, no, it'll never even reach the "fi" because it execs the other process immediately.